server/src/main/java/org/opensearch/extensions/action/TransportActionRequestFromExtension.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions.action;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.transport.TransportRequest;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Transport Action Request from Extension
+ *
+ * @opensearch.api
+ */
+public class TransportActionRequestFromExtension extends TransportRequest {
+    private final String action;
+    private final byte[] requestBytes;
+    private final String uniqueId;
+
+    public TransportActionRequestFromExtension(String action, byte[] requestBytes, String uniqueId) {
+        this.action = action;
+        this.requestBytes = requestBytes;
+        this.uniqueId = uniqueId;
+    }
+
+    public TransportActionRequestFromExtension(StreamInput in) throws IOException {
+        this.action = in.readString();
```

Comment:
I think a `super(in)` is required here.  It definitely doesn't match the `super.writeTo(out)`.

Reply:
Yeah definitely, the tests failed without it. Nice catch :)